doc/specs/#476 - Pinned packages.md

```diff
@@ -0,0 +1,186 @@
+
+The main interface for the pinning feature will be a new command, `winget pin`, with a couple of options:
+
+
+```
+winget pin
+PowerShell: Get-WinGetPinnedPackage
+```
+
+Running `winget pin` with no arguments will show a table of currently pinned packages:
+
+```
+Name                                                 Id                                Version    
+------------------------------------------------------------------------------------------------
+Microsoft Bob                                        Microsoft.Bob                     2.35.0
+iTunes                                               Apple.iTunes                      12.0.199.4
+
+```
+
+```
+winget pin --set <package>
+PowerShell: Set-WinGetPinnedPackage
+```
+
+`<package>` correlates to a query, similar to `winget list`. When running this command, the package information will be stored in the table in the tracking catalog. This command should be able to accept Package Identifers (checked against all repositories) as well as names, tags, etc. If the package does not have a version number locally but is available from a source, then the version number from the source should be used.
+
+If there is no package directly available for software a user wants to pin, but it is [included as part of another package](https://github.com/microsoft/winget-cli/issues/1073), the package which includes said piece of software should be treated as pinned unless a newer version doesn't change the version of the pinned software. 
+
+```
+winget pin --clear <package>
+PowerShell: Remove-WinGetPinnedPackage
+```
+
+It functions similarly to `winget pin --set`, but it clears the pin. This should automatically happen on `winget uninstall`, or if it is detected the package is no longer present on the system.
+
+Enhancements will also need to be made to other commands in the Windows Package Manager, including:
+
+```
+winget upgrade --include-pinned
+PowerShell: Get-WinGetPackage -UpgradeAvailable -Pinned
+```
+
+With the new `--include-pinned` argument, Upgrade should show a separate table in addition to the regular table which shows possible upgrades to pinned packages.
+
+```
+<...>
+Upgrades to pinned packages: 
+
+Name                                                 Id                                Version       Available     Source
+-------------------------------------------------------------------------------------------------------------------------
+Clippy for Azure 2011 Datacenter                     Microsoft.Clippy.2011.Datacenter  2.35.0        2.35.1.2      winget
+```
+
+
+```
+winget <upgrade/install> --include-pinned <package>
+PowerShell: Install-WinGetPackage -Pinned / Upgrade-WinGetPackage -Pinned
+```
+
+Upgrade and Install will now need to use the same argument to bypass the pin for a package temporarily, in which case the pin is updated to pin the new version after the installation completes.
```

Comment:
I don't think that this is necessary.  The goal is that the standard `upgrade --all` ignores these, but specifically targeted packages are how it should be done.  Placing yet another hurdle seems likely to just trip people rather than actually preventing accidental upgrades.

Reply:
Resolved in latest commit. I said that it should warn that you are modifying a pinned package, but continue anyway.